### PR TITLE
Story preview UX

### PR DESCRIPTION
**Pull request by aabounegm**
Supersedes #28. A rebase was needed since this work was done on a branch that received more commits/refactors, and so conflicts were introduced. On the downside, some of the commits might leave the project at an invalid state (in which it wouldn't compile), but this wasn't explicitly checked for. It has only been checked to be compiling and running after the entire rebase was completed.
The rebase also caused me to appear as the **committer** (while @alkaitagi is still the **author**). If you know how to change that (or if it's not a problem for you), let me know.

Description from #28:
> This PR brings needed UX polish into the story previewing screen, as well as some minor UI improvements across several other areas. Also considerably refactored many widgets.

Please confirm that the proposed changes have been moved successfully and nothing was lost in the rebase

### CODE REVIEW

**Comment by alkaitagi** [comment]
Excellent rebase.

**Review by alkaitagi — APPROVED** [review]

File: lib/modules/home/widgets/story_tile.dart [hunk 1]
```diff
@@ -20,7 +20,7 @@ class StoryTile extends StatelessWidget {
         aspectRatio: 1,
         child: Card(
           child: info.imageUrl == null
-              ? const Icon(Icons.landscape_rounded)
+              ? const Icon(Icons.history_edu_rounded)
```
Comment (aabounegm) [inline lib/modules/home/widgets/story_tile.dart:23]: What's the reason behind this icon? That a story is like "history" or what? :D

**Review by aabounegm — COMMENTED** [review]
I have verified that this PR contains essentially what was in the previous one, with the only differences being in merge conflict resolution. I like the file renames as they are more descriptive and better organized. I ran the application and gave it a quick and short test run and didn't face any errors, so we can leave it merged and consider it reviewed even if I haven't gone through it line by line 😁

File: lib/modules/home/widgets/story_tile.dart (hunk 1 above)
Comment (alkaitagi) [reply lib/modules/home/widgets/story_tile.dart:23]: Yes, kind of manuscript of a story. My personal association.

File: lib/modules/editor/screens/actors.dart [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+class ActorsEditorScreen extends StatelessWidget {
+  const ActorsEditorScreen(
+    this.onSelect, {
+    this.selectedId,
+    this.scroll,
+    this.allowNone = false,
+    Key? key,
+  }) : super(key: key);
+
+  final FutureOr<void> Function(Actor?, bool isNew) onSelect;
```
[the remark this answers is not in the document]
Comment (alkaitagi) [reply lib/modules/editor/screens/actors.dart:21]: True. And I'm also sorry that your excellent `Navigator.pop()`-style selection return with `initial` argument doesn't apply here nicely. I somehow feel this PR undid all of your great work on the previous one.

[the remark this answers is not in the document]
Comment (aabounegm) [reply lib/modules/editor/screens/actors.dart:21]: No no, it just built on top of it, and while resolving conflicts I made sure to bring over these changes

**Pull request merged**